learning_resources/etl/loaders.py

```diff
@@ -238,6 +238,43 @@ def load_run(
     return learning_resource_run
 
 
+def load_resource_by_unique_field(
+    platform: LearningResourcePlatform,
+    resource_data: dict,
+    unique_field,
+    unique_value: any,
+) -> tuple[LearningResource, bool]:
+    """
+    Upsert a learning resource using conditional parameters depending
+    on whether the unique field differs from the usual readable_id
+    """
+    readable_id = resource_data.get(READABLE_ID_FIELD)
+    resource_type = resource_data.get("resource_type")
+    if unique_field and unique_field != READABLE_ID_FIELD:
+        # Some dupes may result, so delete any unpublished resources
+        # and update published matching resources with the ingested
+        # readable_ids
+        LearningResource.objects.filter(
+            **{unique_field: unique_value},
+            platform=platform,
+            resource_type=resource_type,
+            published=False,
```

Comment:
This code should be run only when the readable_id is not the `unique_field` value, and in that case if there may be multiple unpublished resources with the same value for `unique_field` that need to be removed to avoid database duplicate errors.  

Good catch on the mitpe courses with no future runs though, those probably should just be skipped altogether instead of (re)created as unpublished.  Not much point in importing unpublished resources.